db/migrate/20150106161639_cleanup_endorsements.rb

```diff
@@ -0,0 +1,11 @@
+class CleanupEndorsements < ActiveRecord::Migration
+  def up
+    Endorsement.find_each(batch_size: 100) do |endo|
```

Comment:
I think `delete_all` would be the way to go since it would be a single SQL query without having to build up any ruby objects. But you'll have to change the query to something like:

```ruby
Endorsement.where(endorsing_user_id: nil).delete_all
```

Not sure if that covers the case where there's a broken `endorsing_user_id` foreign key.